chore(ci): Updating CI to allow tests on docs updates and clean up label logic

castastrophe opened the pull request:
### What has changed and why

- Add PR label/unlabel to automerge task; include ready to merge as a requirement for automerge to work
- Remove pull request review as a firing task for the automerge
- Validate conventional commit syntax before adding the ready to merge label (cc-format.yml)
- Include blocked & needs: & ready: labels before adding the ready to merge label
- Remove paths-ignore from the build task because it can block PRs from merging if they only update documentation
- Separate label workflows to limit the run for the labeler tool to opened & synchronized

### Testing instructions

- [x] Validate labelling
  1. Open a fork branch with this PR as the base.  Commit only a markdown file update.  The build should be kicked off for that PR.
  2. Flag that fork branch PR as `on hold` then change it to draft state.  `work in progress` label should _not_ be added.
  3. Set the `ready: code review` label on the PR and then approve it.  `ready to merge` label should _not_ be added.
  4. Remove the label and then approve the PR again.  The `ready to merge` label should be added.
  5. Confirm that the labelling/unlabelling is not causing an infinite loop with the label task.

### Ready-for-merge Checklist

<!-- Check off items as they are completed.  Feel free to delete items if they are not applicable to your PR. -->

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Repository compiles and tests pass.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

castastrophe commented on .github/workflows/label.yml:
```diff
@@ -10,6 +10,8 @@ on:
       - opened
       - convert_to_draft
       - ready_for_review
+      - labeled
```
@zhawkins I'm not sure about this one - it might cause an infinite loop.  I need to check.

zhawkins reviewed (commented):
Following the testing steps I got to:

> Remove the on hold label. work in progress label should be added.

I've removed the "on hold" label but I don't see a "work in progress" label added. 🤔 

Test PR: https://github.com/patternfly/patternfly-elements/pull/1456

castastrophe commented on .github/workflows/label.yml:
```diff
@@ -1,23 +1,32 @@
+    #   - elements/*/rollup.config.js
+    #   - .github/**/*
+    #   - ./.*
+    #   - ./*.{js,json,yml}
```
I did, I want to circle back to this logic and see how it can work; right now it's blocking merging doc PRs.

castastrophe commented on .github/workflows/tests.yml:
```diff
@@ -11,16 +11,17 @@ on:
+    #   - "generators/**"
+    #   - "**/*.md"
+    #   - "**/*.text"
+    #   - "*.md"
```
Same as label.yml

castastrophe commented:
> I've removed the "on hold" label but I don't see a "work in progress" label added. 🤔

Dang I forgot I'd pulled that out as sort of being out of scope.  I just removed that test case.

zhawkins reviewed (approved):
Updated testing instructions pass 🎉
